contracts/tokens/BaseToken.sol

```diff
@@ -153,5 +164,32 @@ abstract contract BaseToken is Context, ERC721Enumerable, CommunityOwnable {
         super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
     }
 
+    /**
+     * @dev See {IERC1155-safeBatchTransferFrom}.
+     */
+    function safeBatchTransferFrom(address from, address to, uint256[] memory ids, bytes memory data) public virtual {
+        if (from != _msgSender() && !isApprovedForAll(from, _msgSender())) {
+            revert BaseToken_NotAuthorized();
```

Comment:
internally  the ERC721 `transferFrom` is called and it's already checking for ownership and approvals, so if we remove this we save some gas

Reply:
Actually I've used `_safeTransfer()` which ultimately uses `_transfer()` which does no checks on `msg.sender` or `approvals`. 

Thank you for pointing it out. We either keep this check here and use `_safeTransfer()` or, we remove this check here but then have to use `safeTransferFrom()` instead.

I'm gonna change this to use `safeTransferFrom()` and remove the check here.

Thanks again!

Reply:
Updated this as mentioned above.